my-app/src/components/CompanyInformationForm.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import Input from './Input';
+import Textarea from './Textarea';
+import CodeInput from './CodeInput';
+
+export default class CompanyInformationForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { validationStatus: {} };
```

Comment:
po co to zagnieżdżenie i validationStatus skoro masz tylko 1 pole w tym obiekcie?
nie wystarczy:
```js
this.state = {
   // i tutaj bezpośrednio zawartość validationStatus
 }
```